### PR TITLE
feat(ci): Solutions/OSL - upgrade poetry; add default timeout; ignore…

### PR DESCRIPTION
**Pull request by nick-marnik**
… formatting on finicky CI files

The prettier change are to prevent issues like this: https://github.com/ansys/ansys-templates/commit/5439067afacca355f048c3670a85d5388f739e33

@iazehaf @ccpansys I didn't test these changes - do we have any infrastructure ready yet to test this PR to ensure that it can generate the underlying solution and run the CI workflows successfully?

### CODE REVIEW

**Comment by nick-marnik** [comment]
@iazehaf I also wanted to ask what happened to the rest of the CI actions - test, doc, etc.? It looks like they were all removed here, but no explanation was given: https://github.com/ansys/ansys-templates/pull/278

**Comment by iazehaf** [comment]
@nick-marnik I missed [that one](https://github.com/ansys/ansys-templates/pull/390). I just merged this one which fixes the missing bracket issue. Regarding the ci.yml, the idea is to provide a minimum set of jobs but not necessarily all of them. For instance tests and doc actions are not always needed at the beginning of the solution development as tests and doc content are often missing. I am also open to add here all the checks if you think it is better.

File: src/ansys/templates/python/osl_solution/hooks/post_gen_project.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 DESIRED_STRUCTURE = [
-    ".github/workflows/ci.yml",
+    ".github/workflows/build-release.yml",
     ".vscode/launch.json",
```
Comment (iazehaf) [inline src/ansys/templates/python/osl_solution/hooks/post_gen_project.py:10]: @nick-marnik I don't know what is the best naming here, but in the baseline solution we use ci.yml. So we need to come up with a single naming convention.

**Review by iazehaf — APPROVED** [review]
Thanks @nick-marnik ! Just two comments.

File: src/ansys/templates/python/osl_solution/hooks/post_gen_project.py (hunk 1 above)
Comment (nick-marnik) [reply src/ansys/templates/python/osl_solution/hooks/post_gen_project.py:10]: I can't find the previous comment I made, but I have changed this in a few different solutions recently. My argument is that any file in `.github/workflows` is inherently related to CI/CD. My intention is to provide a more descriptive filename that gives some context for what the workflow does.

I akin this to things like `var myVar` or `bool myBool` - it is just repetitive information.

I'm open to suggestions if you think a different filename is better.

**Comment by nick-marnik** [comment]
> @nick-marnik I missed [that one](https://github.com/ansys/ansys-templates/pull/390). I just merged this one which fixes the missing bracket issue. Regarding the ci.yml, the idea is to provide a minimum set of jobs but not necessarily all of them. For instance tests and doc actions are not always needed at the beginning of the solution development as tests and doc content are often missing. I am also open to add here all the checks if you think it is better.

My preference is to add them back. I think that a solution developer could selectively disable jobs that aren't useful, temporarily, by commenting them out.

My biggest concern with the current implementation is: as a solution developer, when I am ready and comfortable adding doc/test, where do I go to get the last YAML code? I don't see it in the template and I doubt anyone will want to go looking for it elsewhere.

If it's ok, I think we could create a separate issue for this because there is a secondary discussion @thesct22 has been involved with related to splitting out build/package/doc/test into different workflow YAML files. Based on the result of that discussion, I think the solution template should come with an out-of-the-box set of default workflows.

**Pull request merged**